python/.pre-commit-config.yaml

```diff
@@ -63,3 +63,11 @@ repos:
       - id: flake8
         args: [ "--ignore=E501,W503,E203" ]
         additional_dependencies: [ flake8-bugbear==22.7.1, flake8-comprehensions==3.10.0 ]
+  - repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.16
+    hooks:
+      - id: mdformat
+        additional_dependencies:
+          - mdformat-black
+          - mdformat-config
+          - mdformat-beautysh
```

Comment:
What's happening here?

Reply:
This is my personal favorite. This will actually detect shell, python, and config blocks in the code, and format it automatically. This way the code, script, and config will be nicely formatted 👍🏻 If it isn't formatted properly, it will fail the CI.

Reply:
Sounds reasonable. In the past, I've just configured those things in mkdocs, but if it's working that's great.